CMakeLists.txt

```diff
@@ -49,10 +49,17 @@ option(WITH_XC_BROWSER "Include browser extension support." OFF)
 # Process ui files automatically from source files
 set(CMAKE_AUTOUIC ON)
 
+set(KEEPASSXC_NIGHTLY_BUILD OFF CACHE BOOL "Set this as a nightly build")
+set(KEEPASSXC_NIGHTLY_VERSION "" CACHE STRING "Special version string for nightly builds")
+
 set(KEEPASSXC_VERSION_MAJOR "2")
 set(KEEPASSXC_VERSION_MINOR "2")
 set(KEEPASSXC_VERSION_PATCH "4")
-set(KEEPASSXC_VERSION "${KEEPASSXC_VERSION_MAJOR}.${KEEPASSXC_VERSION_MINOR}.${KEEPASSXC_VERSION_PATCH}")
+if(KEEPASSXC_NIGHTLY_BUILD)
+  set(KEEPASSXC_VERSION "${KEEPASSXC_NIGHTLY_VERSION}")
```

Comment:
Shouldn't this also start with major.minor.patch and only have another tag at the end? In the Java world it's common to name VCS-HEAD releases x.y.z-SNAPSHOT, for instance.

Reply:
I agree with @phoerious

Reply:
Perhaps we could even reverse the logic of the NIGHTLY flag and replace it with a RELEASE flag. If built without it, it's automatically a snapshot build. That way not only explicit nightlies show the -SNAPSHOT flag, which makes it easier to see if someone built from git or uses an official release.

Reply:
Corrected in last push